cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -401,6 +403,55 @@ public synchronized void updateProperties(NamespaceId namespaceId, NamespaceMeta
       throw new BadRequestException(String.format("Mappings %s for namespace %s cannot be updated once the namespace " +
                                                     "is created.", difference, namespaceId));
     }
+
+    if (namespaceMeta.getRepoConfig() != null) {
+      updateRepoConfig(builder, namespaceMeta.getRepoConfig());
+    }
+    
+    NamespaceMeta updatedMeta = builder.build();
+    nsStore.update(updatedMeta);
+    // refresh the cache with new meta
+    namespaceMetaCache.refresh(namespaceId);
+    LOG.info("Namespace {} updated with meta {}", namespaceId, updatedMeta);
+  }
+
+  private void updateRepoConfig(NamespaceMeta.Builder builder, NamespaceRepositoryConfig repoConfig) {
+    if (repoConfig.getProvider() != null) {
```

Comment:
I would suggest moving this method inside the builder, with a method `setRepositoryConfig(NamespaceRepositoryConfig repoConfig)`